cabot/metricsapp/api/base.py

```diff
@@ -100,74 +86,102 @@ def _add_threshold_data(check, series):
         return series['data']
 
 
+def _points_trigger_high_alert(result, series, series_name, datapoints, check):
+    threshold = check.high_alert_value
+    importance = check.high_alert_importance
+    return _points_trigger_alert(result, series, series_name, datapoints, check, threshold, importance)
+
+
+def _points_trigger_warning(result, series, series_name, datapoints, check):
+    threshold = check.warning_value
+    importance = Service.WARNING_STATUS
+    return _points_trigger_alert(result, series, series_name, datapoints, check, threshold, importance)
+
+
+def _points_trigger_alert(result, series, series_name, datapoints, check, threshold, importance):
+    '''
+    Process a set of points, looking to see if they breach the threshold for the
+    given alert level ('warn' or 'high'). If so, populate the result object and
+    return True. Otherwise return False.
+    '''
+    # If there's no threshold set, then ignore
+    if threshold is None:
+        return False
+
+    # Examine each data point
+    consecutive_failures = 0
+
+    for point in datapoints:
+        timestamp, value = point
+        # If the point fails, increment the consecutive failure count, and
+        # then check if we have enough failed points to return an error.
+        if _point_failure_check(check.check_type, threshold, value):
+            consecutive_failures += 1
+            if consecutive_failures >= check.consecutive_failures:
+                # Set the importance so the check fails at the right level
+                check.importance = importance
+                result.succeeded = False
+                result.error = _get_error_message(check, threshold, importance, series_name, value)
+                result.raw_data = _add_threshold_data(check, series)
```

Comment:
Shouldn't we need to save `check` and `result` here? not sure how this has been working 😱

Reply:
Not sure I follow. Why would we need to save `check` and `result`?

Reply:
Django doesn't update the database unless you call save() https://docs.djangoproject.com/en/2.0/ref/models/instances/#updating-attributes-based-on-existing-fields